src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2240,8 +2240,9 @@ void GameObject::BuildValuesUpdate(uint8 updateType, ByteBuffer* data, Player* t
                 switch (GetGoType())
                 {
                     case GAMEOBJECT_TYPE_QUESTGIVER:
-                        if (ActivateToQuest(target))
-                            dynFlags |= GO_DYNFLAG_LO_ACTIVATE;
+                        // Interaction should only be possible if GO has gossip or is active for quest
+                        if (!ActivateToQuest(target) && !GetGOInfo()->questgiver.gossipID)
+                            dynFlags |= GO_DYNFLAG_LO_NO_INTERACT; // This flag will disable interaction, not setting GO_DYNFLAG_LO_ACTIVATE will not!
```

Comment:
Query: select * from gameobject_template where flags & 4 and type = 2

170 row(s) affected